src/components/editor/page-properties.tsx

```diff
@@ -110,53 +123,55 @@ const PageProperties = ({ pageNumber, surveyId }: Props) => {
     event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
     previousOption: string
   ) => {
-    const updatedOptions = options.map(option => {
+    const updatedOptions = options.map((option) => {
       return option === previousOption ? event.target.value : option;
     });
-  
-    const updatedProperties: PageProperty[] = surveyPages[pageNumber - 1].properties?.some(property => property.type === PagePropertyType.Options)
-      ? surveyPages[pageNumber - 1].properties!.map(property => {
-          return property.type === PagePropertyType.Options
-            ? { ...property, value: JSON.stringify(updatedOptions) }
-            : property;
-        })
+
+    const updatedProperties: PageProperty[] = surveyPages[pageNumber - 1].properties?.some(
+      (property) => property.type === PagePropertyType.Options
+    )
+      ? surveyPages[pageNumber - 1].properties!.map((property) => {
+        return property.type === PagePropertyType.Options
+          ? { ...property, value: JSON.stringify(updatedOptions) }
+          : property;
+      })
       : [
-          ...surveyPages[pageNumber - 1].properties!,
-          {
-            key: PagePropertyType.Options,
-            value: JSON.stringify(updatedOptions),
-            type: PagePropertyType.Options,
-          },
-        ];
-  
+        ...surveyPages[pageNumber - 1].properties!,
+        {
+          key: PagePropertyType.Options,
+          value: JSON.stringify(updatedOptions),
+          type: PagePropertyType.Options
+        }
+      ];
+
     const updatesToPage: Page = {
       ...surveyPages[pageNumber - 1],
-      properties: updatedProperties,
+      properties: updatedProperties
     };
-  
+
     try {
       const updatedPage = await pagesApi.updateSurveyPage({
         pageId: surveyPages[pageNumber - 1].id!,
         surveyId: surveyId,
-        page: updatesToPage,
+        page: updatesToPage
       });
-  
-      const updatedSurveyPages = surveyPages.map(page => (page.id === updatedPage.id ? updatedPage : page));
-  
+
+      const updatedSurveyPages = surveyPages.map((page) =>
+        page.id === updatedPage.id ? updatedPage : page
+      );
+
       setSurveyPages(updatedSurveyPages);
       setOptions(updatedOptions);
     } catch (error) {
-      setError(`${ strings.errorHandling.editSurveysScreen.pageNotSaved }, ${ error }`)    }
+      setError(`${strings.errorHandling.editSurveysScreen.pageNotSaved}, ${error}`);
+    }
   };
-  
 
   /**
    * Adds new question option to question options
    */
   const addNewQuestionOption = () => {
-    options.length
-      ? setOptions([...options, ""])
-      : setOptions([""]);
+    options.length ? setOptions([...options, ""]) : setOptions([""]);
```

Comment:
Is this overly complicated? Wouldn't spreading the empty `options` array yield same result?

Reply:
Fixed